### ISSUE TITLE
Latest Version Won't Work

### ISSUE THREAD

**Issue by dameronw** [DVE2000/Dogbone#26]
I've been using previous versions of Dogbone for ages, and it's worked great (for the most part), but after I updated it thinks for a while, and then the program ends, and nothing has changed to the selected geometry. An item is in the undo section, but no dogbones have been created. Anyone else having any issues?

I've tried deleting and reinstalling.

Any help would be greatly appreciated, as I have hundreds of these to do.

**Comment by dameronw** [comment]
Ok, I've found that if I right click on the feature created in the timeline, edit feature, then it brings up the correct 'cut' - OK, and it cuts out all the dogbones. So it seems to create the correct 'sketch' geometry, or base features used for the cut, but then somehow fails on the 'cut' feature. Anyone else having this?

**Comment by DVE2000** [comment]
Can you export something that has this issue and point me to a download? I can check on it tonight.

**Comment by DVE2000** [comment]
And did you use the non-parameterized version?

**Comment by dameronw** [comment]
I'm not all that familiar with GitHub, but I'll work on uploading something as an example.

Also, I tried both checking and unchecking the 'output unconstrained geometry' box - mostly I use the box checked ie. the faster, non-parametric way.

**Comment by DVE2000** [comment]
Email it to dve2000.2000@gmail.com. I can see if it happens for me and try debug it at least.

And yeah, I'd leave that box checked. The only reason I asked is that slightly different code paths are used depending on that checkbox.

**Comment by DVE2000** [comment]
Also, if you want to get back to the last version that Casey released (before I started adding features), you get get it here: https://github.com/caseycrogers/Dogbone/tree/90d171379c6da3375892f04994729dbf12012637

**Comment by dameronw** [comment]
I re installed the older version of dogbone, and it seems to be working
fine on the design I sent over, albeit with its normal quirks.

If I could have my wishlist of changes to the add in it would be:

Selection filters

   - Currently all versions of the add in, when you select the body you
   wish to dogbone, it finds ALL of the 'up' edges, including edges such as
   where a curved face transitions to a straight face ie a fillet. This is
   generally not a place you want a dogbone. A way to filter these out would
   be incredible. Also a way to filter out all vertical edges which are made
   up of two faces of angle less than X degrees would be great - might kill
   two birds with one stone
   - After the program identifies all vertical edges on a body, I wish it
   would highlight them for you, and let you de-select the ones you don't wish
   to dogbone - this would also help with the above problem
   - If you decide to select edges yourself, rather than the whole body, I
   wish you could filter so that it will ONLY let you select edges, and not
   have to keep selecting through the body, or orbiting around to find the
   line you need, and having to be very specific about which edge you grab to
   not accidentally select the body.

Other

   - When you create a dogbone on a feature which doesn't extend to the top
   of the material, the dogbone also won't go to the top of the material - it
   extends only as long as the edge. Obviously a cnc router cannot do this -
   the dogbone needs to extend all the way to the top of the material. Say
   you've made a dado, which a part fits into, but then you also make a tab on
   the piece which inserts into the dado - this helps with alignment and
   strength, but also, I often work with plywood which is finished on one side
   - having the router cut a shallow dado makes it so that the glue join is
   now wood on wood, not wood on one piece, to the finished surface of the
   other piece.
   - The features you already added were great! choosing direction, minimal
   dogbones, and choosing the 'up' direction was aweosme!

These few features would make this already great add-in even better - it
would be incredible!

Sorry to just list stuff, and I have no idea if you're interested in
working on these or not, but thought I would mention it. Thanks again for
your help!

Will

**Comment by DVE2000** [comment]
Those are good ideas, Will. I'd be happy to work on them, but my biggest issue is time. And I'm a Fusion and Python noob. I will say that for selections, at the moment, you can select all the edges you want first, and THEN run Dogbone. That should help make selections much easier. Because you can set the selection filters and priorities to choose only edges and select through the body as well. I think that takes care of your second two points.

I like the angle selection criterion you suggested in the first point too. I'll look into that one if I can figure out this bug.

That first point under "other" sounds like it shouldn't be hard to do, but I'll be looking at the API reference a lot... So it'll be tough. Noob and all that...

**Comment by DVE2000** [comment]
Ok Will, the bug was introduced in the version that added which allows you to choose ANY direction as up. I still need to track down why the extrudes actually don't happen. But meanwhile, if you use the previous working version, which still has the Y-Up option, you'll be fine: https://github.com/caseycrogers/Dogbone/tree/39c557b3e1c7ac64a05ba5f51b60dfc1b2402305

The above version won't remember your last settings, unfortunately, and I won't get a chance to work on this today. The reason I never saw the bug is probably because I have Z-up as default.

**Comment by dameronw** [comment]
Holy cow. I never tried selecting my edges first, then starting dogbone.
That would have made my life so much easier, for quite a few drawings now.
My life just got better - thanks! It's funny how the slightest change in
workflow in Fusion can make such a huge difference.

Also, thanks for working on those ideas - I know what you mean about time.
If I had the time, I'd love to try to learn to do it myself - the coding
appeals to me. But, there is just no way that is going to happen anytime
soon. Maybe one day next year though.

Will

**Comment by DVE2000** [comment]
I found the issue. I'd added an feature to limit the dogbones to selected bodies or bodies of selected edges. This was because another user had an assembly and started getting dogbone holes being cut into bodies touching the body where the dogbones where actually being created. For some unknown reason, this prevented the dogbones from being cut on your design. I stepped through the code with the debugger, and I really think this is some sort of Fusion bug. Dunno.

In any event, limiting participation to selected bodies etc is now a checkbox option. The latest version should work fine for you, Will. Just leave that checkbox unchecked!

I'm going to close this bug.
